shell/browser/browser_linux.cc

```diff
@@ -147,58 +147,64 @@ bool Browser::IsEmojiPanelSupported() {
-  }
-  if ((str = opts.FindStringKey("copyright"))) {
-    gtk_about_dialog_set_copyright(dialog, str->c_str());
-  }
-  if ((str = opts.FindStringKey("website"))) {
-    gtk_about_dialog_set_website(dialog, str->c_str());
-  }
-  if ((str = opts.FindStringKey("iconPath"))) {
-    GError* error = nullptr;
-    constexpr int width = 64;   // width of about panel icon in pixels
-    constexpr int height = 64;  // height of about panel icon in pixels
-
-    // set preserve_aspect_ratio to true
-    GdkPixbuf* icon =
-        gdk_pixbuf_new_from_file_at_size(str->c_str(), width, height, &error);
-    if (error != nullptr) {
-      g_warning("%s", error->message);
-      g_clear_error(&error);
-    } else {
-      gtk_about_dialog_set_logo(dialog, icon);
-      g_clear_object(&icon);
+  if ((opts.is_dict())) {
+    if ((str = opts.FindStringKey("applicationName"))) {
+      gtk_about_dialog_set_program_name(dialog, str->c_str());
     }
-  }
-
-  if ((val = opts.FindListKey("authors"))) {
-    std::vector<const char*> cstrs;
-    for (const auto& authorVal : val->GetList()) {
-      if (authorVal.is_string()) {
-        cstrs.push_back(authorVal.GetString().c_str());
+    if ((str = opts.FindStringKey("applicationVersion"))) {
+      gtk_about_dialog_set_version(dialog, str->c_str());
+    }
+    if ((str = opts.FindStringKey("copyright"))) {
+      gtk_about_dialog_set_copyright(dialog, str->c_str());
+    }
+    if ((str = opts.FindStringKey("website"))) {
+      gtk_about_dialog_set_website(dialog, str->c_str());
+    }
+    if ((str = opts.FindStringKey("iconPath"))) {
+      GError* error = nullptr;
+      constexpr int width = 64;   // width of about panel icon in pixels
+      constexpr int height = 64;  // height of about panel icon in pixels
+
+      // set preserve_aspect_ratio to true
+      GdkPixbuf* icon =
+          gdk_pixbuf_new_from_file_at_size(str->c_str(), width, height, &error);
+      if (error != nullptr) {
+        g_warning("%s", error->message);
+        g_clear_error(&error);
+      } else {
+        gtk_about_dialog_set_logo(dialog, icon);
+        g_clear_object(&icon);
       }
     }
-    if (cstrs.empty()) {
-      LOG(WARNING) << "No author strings found in 'authors' array";
-    } else {
-      cstrs.push_back(nullptr);  // null-terminated char* array
-      gtk_about_dialog_set_authors(dialog, cstrs.data());
+
+    if ((val = opts.FindListKey("authors"))) {
+      std::vector<const char*> cstrs;
+      for (const auto& authorVal : val->GetList()) {
+        if (authorVal.is_string()) {
+          cstrs.push_back(authorVal.GetString().c_str());
+        }
+      }
+      if (cstrs.empty()) {
+        LOG(WARNING) << "No author strings found in 'authors' array";
+      } else {
+        cstrs.push_back(nullptr);  // null-terminated char* array
+        gtk_about_dialog_set_authors(dialog, cstrs.data());
+      }
     }
+  } else {
+    LOG(WARNING) << "Called showAboutPanel(), but didn't use "
```

Comment:
Unclear what we do on other platforms but should this throw an error?

Also should we short-circuit this and not even allocate the GtkWidget if no options are set?

Reply:
I'm in agreement w the above: this should be short-circuited i would say.

Reply:
> Unclear what we do on other platforms but should this throw an error?
* On macOS, I believe there are defaults based on the `.plist` file.
* Windows behaviour is TBD, as the PR is still open (#19420).

> Also should we short-circuit this and not even allocate the GtkWidget if no options are set?
So just not display the panel at all? I could make that happen.